key.core/src/main/java/org/key_project/proof/LocationVariableTracker.java

```diff
@@ -0,0 +1,54 @@
+/* This file is part of KeY - https://key-project.org
+ * KeY is licensed under the GNU General Public License Version 2
+ * SPDX-License-Identifier: GPL-2.0-only */
+package org.key_project.proof;
+
+import java.util.Map;
+import java.util.WeakHashMap;
+
+import de.uka.ilkd.key.logic.OpCollectorJavaBlock;
+import de.uka.ilkd.key.logic.op.LocationVariable;
+import de.uka.ilkd.key.proof.Proof;
+import de.uka.ilkd.key.proof.ProofEvent;
+import de.uka.ilkd.key.proof.RuleAppListener;
+import de.uka.ilkd.key.rule.RuleApp;
+
+public class LocationVariableTracker implements RuleAppListener {
```

Comment:
Can you please create a bit of JavaDoc here? It should be clear also without the context of this PR what the class is/can be used for.